changelog/@unreleased/pr-684.v2.yml

```diff
@@ -0,0 +1,16 @@
+type: break
+break:
+  description: "## Before this PR\n`gitVersion()` or `versionDetails()` does not have
+    caching, every time it is called, it will create a new 'VersionDetails' object
+    and call the expensive `git describe` operation to get the version. This causes
+    long configuration time in a lot of the repos where `version gitVersion()` is
+    called under allprojects. \n\n## After this PR\nBuild service is used so if we
+    call `gitVersion()` on all projects, we will not call the expensive `git describe`
+    many times. Instead, the build service will have a concurrent hash map that stores
+    the created `VersionDetails` object with a specific prefix as key the first time
```

Comment:
This sentence is too much detail for a user to care about - I'd leave out any details of how the caching happens and just say it's cached.